dev/hooks/pre-commit

```diff
@@ -4,7 +4,7 @@ BOLD='\033[1;36m'
 NC='\033[0m'
 printf "${BOLD}[PRE-COMMIT]${NC} formatting code...\n"
 cargo +nightly fmt --all
-git add -u
```

Comment:
The reason this is here is to restage files that were affected by cargo format, removing it has the chance that you forget to stage files that were formatted. Perhaps there is a way to find out which files were affected by the format and stage only those.

I would leave this in for now

Reply:
I think switching to [pre-commit](https://pre-commit.com/) is the best choice here. Its quite easy to setup, extensible and automatically handles cases like this

Reply:
I'll take a look at this. If possible I would like to avoid additional dependencies for developers however